ingestion/src/legacy/raw.rs

```diff
@@ -110,26 +112,18 @@ fn parse_obs(
 
             let (sensor, level) = col.sensor_and_level.unwrap_or((0, 0));
 
-            let val = val.trim();
-
-            // // TODO: is this right?
-            // // Ignore empty values
-            if val.is_empty() {
-                continue;
-            }
-
             let param = match param_entry {
                 Some(entry) => Param::Id(entry.id),
-                None => Param::Code(col.param_code),
+                None => Param::Code(col.param_code.clone()),
             };
 
-            let value: ObsType = if param_entry.is_some() && !param_entry.unwrap().is_scalar {
-                ObsType::NonScalar(val.to_string())
+            let value: ObsType = if (param_entry.is_some() && !param_entry.unwrap().is_scalar)
+                // things marked as scalar in stinfosys that are known not to be floats
+                || kldata::SPECIAL_CASES.contains(&col.param_code.as_str())
+            {
+                parse_nonscalar(val)
```

Comment:
I think this if condition should be reversed, it tries to parse a scalar even when there's no param, giving rise to wrong logs.

```suggestion
            let value: ObsType = if param_entry.is_some() && param_entry.unwrap().is_scalar
                // things marked as scalar in stinfosys that are known not to be floats
                && !kldata::SPECIAL_CASES.contains(&col.param_code.as_str())
            {
                parse_scalar(val)
```